packages/hardhat-core/src/internal/hardhat-network/provider/modules/hardhat.ts

```diff
@@ -385,6 +389,24 @@ export class HardhatModule {
     return validateParams(params, optional(rpcQuantity), optional(rpcQuantity));
   }
 
+  // hardhat_dumpState
+  private _dumpStateParams(params: any[]): [] {
+    return validateParams(params);
+  }
+
+  private async _dumpStateAction() {
+    return this._node.dumpState();
```

Comment:
Wouldn't this return a HUGE buffer?

Reply:
for synthetix integration tests, we measure this to come out to about 1 megabyte. So yes, a pretty big buffer, but I think the RPC can handle this?

I can't really think of a much better way to do it otherwise.

Reply:
probably would be good to write an integration test for this case. Is there a guide on how to do this with hardhat?